src/main/java/com/example/weather/client/wrapper/HttpRequestWrapperImpl.java

```
@@ -0,0 +1,23 @@
+package com.example.weather.client.wrapper;
+
+import lombok.SneakyThrows;
+import org.springframework.http.HttpRequest;
+import org.springframework.http.client.support.HttpRequestWrapper;
+
+import java.net.URI;
+
+public class HttpRequestWrapperImpl extends HttpRequestWrapper {
+
+    private final String keyValue;
+
+    public HttpRequestWrapperImpl(HttpRequest request, String keyValue) {
+        super(request);
+        this.keyValue = keyValue;
+    }
+
+    @SneakyThrows
```

in this case @SneakyThrows would swallow the error, and you wouldn't know what happened basing on a stack trace